File: ReferenceWRLImplementation/CSharpUnitTestProject/StaticClassUnitTests.cs
using Microsoft.VisualStudio.TestPlatform.UnitTestFramework;
using ReferenceWRLImplementation;

namespace CSharpUnitTestProject
{
    [TestClass]
    public class StaticClassUnitTests
    {
        [TestMethod]
        public void StaticMethodWithNoParametersTest()
        {
            StaticClass.StaticMethodWithNoParameters();
        }

        [TestMethod]
        public void StaticMethodWithParamTest()
        {
            StaticClass.StaticMethodWithParam(20);
        }

        [TestMethod]
        public void StaticMethodWithRetValTest()
        {
            uint expectedVal = 25;
            var actualVal = StaticClass.StaticMethodWithRetVal();
            Assert.AreEqual(expectedVal, actualVal);
        }
    }
}

File: ReferenceWRLImplementation/CXUnitTestApp/StaticClassUnitTests.cpp
#include "pch.h"
#include "CppUnitTest.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace ReferenceWRLImplementation;

namespace CXUnitTestApp
{
    TEST_CLASS(StaticClassUnitTests)
    {
    public:
        TEST_METHOD(StaticMethodWithNoParametersCPPTest)
        {
            StaticClass::StaticMethodWithNoParameters();
        }

        TEST_METHOD(StaticMethodWithParamCPPTest)
        {
            StaticClass::StaticMethodWithParam(20);
        }

        TEST_METHOD(StaticMethodWithRetValCPPTest)
        {
            UINT32 expectedVal = 25;
            auto actualVal = StaticClass::StaticMethodWithRetVal();
            Assert::AreEqual(expectedVal, actualVal);
        }
    };
}